### PR TITLE
Added the UserStatusChangeProcessor and its implementation

### PR DESCRIPTION
**Pull request by delvh**
this branch serves as solution to the merge conflict between
f/user_status_change_processor and develop

### CODE REVIEW

File: src/main/java/envoy/client/net/UserStatusChangeProcessor.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	public UserStatusChangeProcessor(Client client) { this.client = client; }
+
+	@Override
+	public void accept(UserStatusChangeEvent evt) {
+		logger.info("Received " + evt);
+		List<User> contacts = client.getContacts().getContacts();
+		contacts.stream().filter((user) -> user.getId() == evt.getId()).findFirst().get().setStatus(evt.get());
+		EventBus.getInstance().dispatch(evt);
+	}
+
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/client/net/UserStatusChangeProcessor.java]: This empty line is unnecessary.

File: src/main/java/envoy/client/net/UserStatusChangeProcessor.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Project: <strong>envoy-client</strong><br>
+ * File: <strong>UserStatusChangeProcessor.java</strong><br>
+ * Created: <strong>2 Feb 2020</strong><br>
+ *
+ * @author Leon Hofmeister
+ * @since Envoy v0.3-alpha
+ */
+public class UserStatusChangeProcessor implements Consumer<UserStatusChangeEvent> {
+
+	private static final Logger	logger	= EnvoyLog.getLogger(UserStatusChangeProcessor.class.getSimpleName());
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/client/net/UserStatusChangeProcessor.java]: Usually, the `Logger` is declared below instance variables and separated with an empty line.

File: src/main/java/envoy/client/net/UserStatusChangeProcessor.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	 * @param client the {@link Client} who receives an
+	 *               {@link UserStatusChangeEvent}
+	 * @since Envoy v0.3-alpha
+	 */
+	public UserStatusChangeProcessor(Client client) { this.client = client; }
+
+	@Override
+	public void accept(UserStatusChangeEvent evt) {
+		logger.info("Received " + evt);
+		List<User> contacts = client.getContacts().getContacts();
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/client/net/UserStatusChangeProcessor.java]: I suppose we do not need the temporary variable and this and the next line into one statement.

File: src/main/java/envoy/client/ui/ChatWindow.java [hunk 4]
```diff
@@ -211,6 +212,9 @@ public void keyReleased(KeyEvent e) {
 		// Listen to theme changes
 		EventBus.getInstance().register(ThemeChangeEvent.class, (evt) -> applyTheme((Theme) evt.get()));
 
+		// Listen to UserStatus changes
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/client/ui/ChatWindow.java]: The current style seems to be `// Listen to user status changes`.

**Comment by delvh** [comment]
kann ich mergen?

**Comment by CyB3RC0nN0R** [comment]
> can I merge?

Yes, you can.

**Pull request merged**